README.md

```diff
@@ -1 +1,62 @@
-# fastify-deprecation
\ No newline at end of file
+# fastify-warning
+
+[![js-standard-style](https://img.shields.io/badge/code%20style-standard-brightgreen.svg?style=flat)](http://standardjs.com/)  ![build](https://github.com/fastify/fastify-warning/workflows/build/badge.svg)
+
+A small utility, used by Fastify itself, for generating consistent warning objects across your codebase and plugins.
+It also exposes a utility for emitting those warnings, guaranteeing that they are issued only once.
+
+### Install
+```
+npm i fastify-warning
+```
+
+### Usage
+
+The module exports a builder function that returns an utility for creating warning and emitting them.
+
+```js
+const warning = require('fastify-warning')()
+```
+
+```
+warning.create(name, code, message)
```

Comment:
For some time now I have been really enamored with named parameters. What do you think of this signature instead?:

```suggestion
warning.create({ name, code, message })
```

Reply:
I like to use options objects as parameters when some of those parameters are optional, given that here all of them are mandatory, I think we could keep them as is. Also, this "style" follows the same as the `fastify-error` package.

Reply:
I think they give clarity to the reader. For example, random bit of code:

```js
const warning = fastifyWarning.create('foo', 'fst_foo', 'a warning')
```

vs


```js
const warning = fastifyWarning.create({ name: 'foo', code: 'fst_foo', message: 'a warning' })
```

In my opinion, the second is easier to read by anyone, not just those who are already familiar with the API.